src/lib/coda_base/signature.ml

```diff
@@ -3,81 +3,87 @@
 
 open Core
 open Module_version
-open Snark_params.Tick
+open Tick_intf
 
-module Stable = struct
-  module V1 = struct
-    module T = struct
-      type t = Field.t * Inner_curve.Scalar.t
-      [@@deriving sexp, eq, compare, hash, bin_io, version {asserted}]
+module Make (Tick : TICK) = struct
+  module Stable = struct
+    module V1 = struct
+      module T = struct
+        type t = Tick.Field.t * Tick.Inner_curve.Scalar.t
+        [@@deriving sexp, eq, compare, hash, bin_io, version {asserted}]
```

Comment:
The linter should be giving a warning about `bin_io` and `version` inside functors.

The solution is to define a type with parameters outside the functor. There's an example in `vrf_lib`/standalone.ml`. We should be using `%%versioned_asserted` for that (the registration functors will be going away, eventually).

Reply:
This case is tricky, because of the call to `Make_base58_check`, which requires a `bin_io` type. You may have to put that where the functor is applied. I had to do something like that for `Non_zero_curve_point` (it was painful).